src/classes/RD2_DataMigration_BATCH.cls

```diff
@@ -0,0 +1,217 @@
+            .withRelatedOpportunities(opps);
+
+        npe03__Recurring_Donation__c convertedRD = migration.convertToEnhancedRD();
+
+        if (convertedRD != null) {
+            convertedRDs.add(convertedRD);
+
+            rdSchedules.addAll(migration.getScheduleRecords());
+
+            // TODO Opportunity Updates
+        }
+        // TODO Retrieve any validation errors that blocked the conversion
+    }
+
+    /**
+    * @description When all RD's have been processed, handle any required DML and error/exception logging
+    * @param sp SavePoint if a rollback is required for dry run mode
+    */
+    private void handleDmlAndLogging(Savepoint sp) {
+        // TODO Rewrite this to capture and log dml exceptions in a new logging service
+        Database.update(convertedRDs, false);
+
+        // TODO If there is an error updating an RD, do not insert the schedule or update Opps for that RD
+        Database.insert(rdSchedules, false);
+
+        if (this.dryRunMode) {
+            Database.rollback(sp);
+        }
+
+        // TODO Commit any logging records after the rollback
+    }
+
+    /**
+    * @description Build the query of all standard RD fields to use for the batch query
+    * @return SOQL
+    */
+    private String getRDMigrationQuery() {
+        Set<String> rdFields = new Set<String>{
+            'Id',
+            'Name',
+            'npe03__Open_ended_status__c',
+            'npe03__Next_payment_date__c',
+            'npe03__Paid_Amount__c',
+            'npe03__Organization__c',
+            'npe03__Contact__c',
+            'npe03__Installment_amount__c',
+            'npe03__Installments__c',
+            'npe03__Amount__c',
+            'npe03__Total__c',
+            'npe03__Installment_period__c',
+            'npe03__Date_established__c',
+            'npe03__Schedule_type__c',
+            'npe03__Recurring_donation_campaign__c',
+            'npe03__Total_paid_installments__c',
+            'Ownerid',
+            'Always_use_last_day_of_month__c',
+            'Day_of_Month__c',
+            'StartDate__c',
+            'InstallmentFrequency__c',
+            'PaymentMethod__c',
+            'Status__c',
+            'RecurringType__c'
+        };
+        if (UserInfo.isMultiCurrencyOrganization()) {
+            rdFields.add('CurrencyIsoCode');
+        }
+
+        UTIL_Query queryBuilder = new UTIL_Query()
+            .withFrom(npe03__Recurring_Donation__c.SObjectType)
+            .withSelectFields(rdFields)
+            .withOrderBy('npe03__Last_Payment_Date__c');
+
+        return queryBuilder.build();
+    }
+
+    /**
+    * @description Build a map of related Opportunities for each recurring donation
+    * @param rds Collection of Recurring Donations to retrieve related Opps for
+    * @return Map<Id, List<Opportunity>>
+    */
+    public Map<Id, List<Opportunity>> getRelatedOpportunities(List<npe03__Recurring_Donation__c> rds) {
```

Comment:
`private` if not used outside the class.

Reply:
Good catch